cmd/acraserver/listener.go

```diff
@@ -107,30 +145,149 @@ func (server *SServer) handleConnection(connection net.Conn) {
 
 // start listening connections from proxy
 func (server *SServer) Start() {
-	listener, err := network.Listen(server.config.GetAcraConnectionString())
+	var connection net.Conn
+	var listener, err = network.Listen(server.config.GetAcraConnectionString())
 	if err != nil {
 		log.WithError(err).Errorln("can't start listen connections")
+		ErrorSignalChannel <- syscall.SIGTERM
 		return
 	}
-	defer listener.Close()
+	server.listenerACRA = listener
 	server.addListener(listener)
+
+	log.Infof("start listening %s", server.config.GetAcraConnectionString())
+	for {
+		connection, err = listener.Accept()
+		if err != nil {
+			if nerr, ok := err.(net.Error); ok && nerr.Timeout() {
+				log.Infoln("stop accepting new connections due net.Timeout")
+				return
+			}
+			log.WithError(err).Errorf("can't accept new connection (connection=%v)", connection)
+		}
+		// unix socket and value == '@'
+		if len(connection.RemoteAddr().String()) == 1 {
+			log.Infof("new connection to acraserver: <%v>", connection.LocalAddr())
+		} else {
+			log.Infof("new connection to acraserver: <%v>", connection.RemoteAddr())
+		}
+		go func() {
+			server.cmACRA.Add(1)
```

Comment:
what mean `1`?

Reply:
oh, it's incrementing counter. Would be more readable, if the call will look like `server.cmAcra.Incr()`

Reply:
done